pyro/nn/auto_reg_nn.py

```diff
@@ -183,4 +198,17 @@ def forward(self, x):
         if self.skip_layer is not None:
             h = h + self.skip_layer(x)
 
-        return h
+        # Shape the output into self.count_params of dimension self.param_dim*self.input_dim
+        if self.count_params == 1:
+            return h
+        else:
+            params = [h[..., (i * self.param_dim * self.input_dim):((i + 1) * self.param_dim * self.input_dim)]
```

Comment:
can you use `torch.unbind` or the like here?

Reply:
It's seems quite complicated because there are several cases - see the latest commit. `torch.unbind` can be used for the case where all the parameters have the same dimension, but otherwise I think you need to extract the parameters with slices